compiler/resolution/callDestructors.cpp

```diff
@@ -614,9 +618,20 @@ void ReturnByRef::transformMove(CallExpr* moveExpr)
     // But... if we're replacing an init copy, we got to a user
     // variable, so add an unalias call if there is one
     if (rhsFn->hasFlag(FLAG_INIT_COPY_FN)) {
-      FnSymbol* unaliasFn = getUnalias(useLhs->type);
       if (unaliasFn) {
-        callExpr->insertAfter(new CallExpr(unaliasFn, refVar));
+        VarSymbol* unaliasTemp = newTemp("unaliasTemp", unaliasFn->retType);
+        callExpr->insertBefore(new DefExpr(unaliasTemp));
+        callExpr->insertAfter(new CallExpr(PRIM_MOVE, unaliasTemp, new CallExpr(unaliasFn, refVar)));
+
+        // Replace all uses of 'useLhs' with 'unaliasTemp', except for the
+        // addr-of case we just created.
+        for_SymbolSymExprs(se, useLhs) {
```

Comment:
While I like for_SymbolSymExprs, in this case I think it'd be better to move this logic up.
Instead of removing the redundant init/auto copy and then adding back in an unalias call,
let's replace the redundant init copy with unalias in the appropriate cases.

I'm talking about the line

```
 copyExpr->replace(copyExpr->get(1)->remove());
```